psyneulink/core/llvm/codegen.py

```diff
@@ -260,6 +344,36 @@ def visit_Name(self, node):
 
     def visit_Attribute(self, node):
         val = self.visit(node.value)
+
+        # special case numpy attributes
+        if node.attr == "shape":
+            shape = helpers.get_array_shape(val)
+            return ir.LiteralStructType([self.ctx.float_ty] * len(shape))(shape)
+        elif node.attr == "astype":
+            def astype(ty):
+                def _convert(ctx, builder, x):
+                    if helpers.is_pointer(x):
+                        x = builder.load(x)
+                    if helpers.is_integer(x) and ty is ctx.float_ty:
+                        if helpers.is_boolean(x):
+                            return builder.uitofp(x, ty)
+                        return builder.sitofp(x, ty)
+                    elif helpers.is_floating_point(x) and ty is self.register["int"]:
+                        return builder.fptosi(x, ty)
+                    elif (helpers.is_floating_point(x) and ty is ctx.float_ty) or \
+                         (helpers.is_integer(x) and ty is self.register["int"]):
+                        return x
+                    else:
+                        raise Exception(f"Tried to convert {x} to invalid type: {ty}")
```

Comment:
No runtime exceptions. They indicate error to the user, there's nothing in the compiler that the user should care about.